go/vt/utils/utils.go

```diff
@@ -16,11 +16,13 @@ limitations under the License.
 
 package utils
 
+import querypb "github.com/youtube/vitess/go/vt/proto/query"
+
 // utils.go contains general utility functions used in the splitquery package.
 
 // CloneBindVariables returns a shallow-copy of the given bindVariables map.
-func CloneBindVariables(bindVariables map[string]interface{}) map[string]interface{} {
-	result := make(map[string]interface{}, len(bindVariables))
+func CloneBindVariables(bindVariables map[string]*querypb.BindVariable) map[string]*querypb.BindVariable {
```

Comment:
Actually all values inside bind variables and sqltypes.Value are read-only. So, for all practical purposes, the shallowness is not material. This was the reason to implement the convenience functions for them, to disallow (discourage) direct manipulation of those fields.

So, Clone, Copy, or CopyMap shouldn't matter. Maybe I'll still rename to Copy, because that's more natural as a name.